python/ray/data/dataset_iterator.py

```diff
@@ -239,6 +241,14 @@ def stats(self) -> str:
         """Returns a string containing execution timing information."""
         raise NotImplementedError
 
+    @property
+    @abc.abstractmethod
+    @DeveloperAPI
```

Comment:
Should this be DeveloperAPI? I don’t think we expect library developers to be implementing this method.

this should just be private I feel.

Reply:
Private makes sense to me.

Fixed.